api/src/main/java/org/openmrs/ui/framework/fragment/FragmentFactory.java

```diff
@@ -1,576 +1,600 @@
+		possibleArguments.put(PageRequest.class, context.getPageContext().getRequest());
+		possibleArguments.put(HttpServletRequest.class, context.getPageContext().getRequest().getRequest());
+        possibleArguments.put(HttpSession.class, context.getPageContext().getRequest().getRequest().getSession());
+		possibleArguments.put(FragmentFactory.class, this);
+		possibleArguments.put(UiUtils.class, new FragmentUiUtils(context));
+		possibleArguments.put(Session.class, context.getPageContext().getRequest().getSession());
+		possibleArguments.put(ApplicationContext.class, applicationContext);
+		possibleArguments.put(ServletContext.class, servletContext);
+        if (possibleFragmentControllerArgumentProviders != null) {
+            for (PossibleFragmentControllerArgumentProvider provider : possibleFragmentControllerArgumentProviders) {
+                provider.addPossibleFragmentControllerArguments(possibleArguments);
+            }
+        }
+
+        String httpRequestMethod = context.getPageContext().getRequest().getRequest().getMethod();
+        return UiFrameworkUtil.executeControllerMethod(context.getController(), httpRequestMethod, possibleArguments, conversionService, applicationContext);
+	}
+
+    /**
+	 * @param request
+	 * @param viewName if not null, overrides what is specified in request
+	 * @return
+	 * @should get a view from the requested provider
+	 * @should get a view from any provider if none is specified
+	 * @should fail if an invalid provider name is specified
+	 */
+	FragmentView getView(FragmentRequest request, String viewName) {
+		if (viewName == null)
+			viewName = request.getMappedFragmentId();
+		String providerAndFragmentId = request.getMappedProviderName() + ":" + viewName;
+		if (!isDevelopmentMode()) {
+			if (viewCache.containsKey(providerAndFragmentId)) {
+				return viewCache.get(providerAndFragmentId);
+			}
+		}
+		if ("*".equals(request.getMappedProviderName())) {
+			for (FragmentViewProvider p : viewProviders.values()) {
+				FragmentView ret = p.getView(viewName);
+				if (ret != null) {
+					if (!isDevelopmentMode())
+						viewCache.put(providerAndFragmentId, ret);
+					return ret;
+				}
+			}
+		} else {
+			FragmentViewProvider provider = viewProviders.get(request.getMappedProviderName());
+			if (provider == null) {
+				throw new UiFrameworkException("No view provider: " + request.getMappedProviderName());
+			}
+			return provider.getView(viewName);
+		}
+		return null;
+	}
+	
+	public Object getController(String providerName, String fragmentName) {
+		FragmentRequest request = new FragmentRequest(providerName, fragmentName);
+		return getController(request);
+	}
+	
+	/**
+	 * Allow modules to override fragment handling via {@link FragmentRequestMapper}s.
+	 * Sets this internal fragment provider and fragmentId on request
+	 * @param request
+	 */
+	private FragmentRequest mapInternalFragmentId(FragmentRequest request) {
```

Comment:
I assume that you copy-paste-edited this name from PageFactory, but I don't actually like it. (I don't know what I was thinking years ago...). Can you rename this method to "maybeOverrideFragmentProviderAndId"?